src/pages/api/setRole.ts

```diff
@@ -0,0 +1,28 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { UserRecord } from 'firebase-admin/auth'
+
+import { auth, firestore } from '@/lib/temp/firebase/admin/init'
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { email, roles } = JSON.parse(req.body)
+  const claims: Record<string, boolean> = { ...roles }
+  try {
+    const userRecord: UserRecord = await auth.getUserByEmail(email)
+    // See the UserRecord reference doc for the contents of userRecord.
+    await auth.setCustomUserClaims(userRecord.uid, claims)
+    // const userRecord: UserRecord = await auth.getUser(userId)
+    const doc: Record<string, boolean> = { ...roles, by: 'none' }
+    await firestore
+      .collection('roles')
+      .doc(userRecord.email || userRecord.uid)
```

Comment:
We're happy to name the document by either field? Could lead to duplicates

Reply:
im using uid for now and will change it when im implementing the user invite feature